lib/omniauth/strategies/saml.rb

```diff
@@ -103,15 +109,23 @@ def other_phase
       uid { @name_id }
 
       info do
-        {
-          :name  => @attributes[:name],
-          :email => @attributes[:email] || @attributes[:mail],
-          :first_name => @attributes[:first_name] || @attributes[:firstname] || @attributes[:firstName],
-          :last_name => @attributes[:last_name] || @attributes[:lastname] || @attributes[:lastName]
-        }
+        found_attributes = options.attribute_statements.map do |key, values|
+          attribute = find_attribute_by(values)
+          [key, attribute]
```

Comment:
Hmm...the other option would be to treat it as a validation failure and raise an exception.  That seems overly aggressive, but admittedly I have very little experience with environments in which SAML is used.  There may very well be some environments in which it makes sense to treat a response with a "missing" Attribute as invalid.